docs/resources/hyperdrive_config.md

```diff
@@ -0,0 +1,90 @@
+---
+page_title: "cloudflare_hyperdrive_config Resource - Cloudflare"
+subcategory: ""
+description: |-
+  The Hyperdrive Config https://developers.cloudflare.com/hyperdrive/ resource allows you to manage Cloudflare Hyperdrive Configs.
+---
+
+# cloudflare_hyperdrive_config (Resource)
+
+The [Hyperdrive Config](https://developers.cloudflare.com/hyperdrive/) resource allows you to manage Cloudflare Hyperdrive Configs.
+
+## Example Usage
+
+```terraform
+# Config using all default values
+resource "cloudflare_hyperdrive_config" "no_defaults" {
+  account_id = "f037e56e89293a057740de681ac9abbe"
+  name       = "my-hyperdrive-config"
+  password   = "my-password"
+  origin     = {
+    database = "postgres"
+    host     = "my-database.example.com"
+  }
+}
+
+# Config not using any default values
+resource "cloudflare_hyperdrive_config" "no_defaults" {
+  account_id = "f037e56e89293a057740de681ac9abbe"
+  name       = "my-hyperdrive-config"
+  password   = "my-password"
+  origin     = {
+    database = "postgres"
+    host     = "my-database.example.com"
+    port     = 5432
+    scheme   = "postgres"
+    user     = "my-user"
+  }
+  caching    = {
+    disabled               = false
+    max_age                = 60
+    stale_while_revalidate = 30
+  }
+}
+```
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `account_id` (String) The account identifier to target for the resource.
+- `name` (String) The name of the Hyperdrive configuration.
+- `origin` (Attributes) The origin details for the Hyperdrive configuration. (see [below for nested schema](#nestedatt--origin))
+- `password` (String) The password of the Hyperdrive configuration.
+
+### Optional
+
+- `caching` (Attributes) The caching details for the Hyperdrive configuration. (see [below for nested schema](#nestedatt--caching))
+- `id` (String) The identifier of this resource. This is the hyperdrive config value.
+
+<a id="nestedatt--origin"></a>
+### Nested Schema for `origin`
+
+Required:
+
+- `database` (String) The name of your origin database.
+- `host` (String) The host (hostname or IP) of your origin database.
+
+Optional:
+
+- `port` (Number) The port (default: 5432 for Postgres) of your origin database.
```

Comment:
These are all required by the API. When implementing the `wrangler` and UI interfaces we just supply the default when not entered by the user. You could do the same here, but you will get `400 Bad Request` back from the API otherwise.

Reply:
Gotcha, thanks @OilyLime, does the same for for the `caching` section too?

Reply:
Oh also, that means we probably need further updates [here also](https://github.com/cloudflare/cloudflare-go/pull/1501)?

Reply:
> does the same for for the caching section too?
No, those are optional. They do have constraints, but it would be overkill (and cause drift) to guard against those constraints here.

> need further updates https://github.com/cloudflare/cloudflare-go/pull/1501?
I'll leave some comments there, thanks.